### PR TITLE
chore: set max_node_depth default and add messages in search_paths

### PR DESCRIPTION
**Pull request by atsushi421**
Signed-off-by: atsushi421 <a.yano.578@ms.saitama-u.ac.jp>

This PR set a default value of `max_node_depth` and add messages in `Architecture.search_paths()`

 - JIRA ticket: https://tier4.atlassian.net/browse/T4PB-20205
 - To reproduce: https://drive.google.com/drive/folders/1Qd3BprOFOVDHp3b0_1Sf9snEFfcJzL4W?usp=share_link

### CODE REVIEW

File: src/caret_analyze/architecture/architecture.py [hunk 1]
```diff
@@ -185,10 +185,42 @@ def search_paths(
...
+        max_node_depth = max_node_depth or default_depth
+
+        # Print message before search
+        msg_recommend_intermediate = (
+            'If the paths you want to measure cannot be found, '
+            'consider specifying intermediate nodes. '
+        )
+        msg_detail_page = (
+            'For details, '
+            'see https://tier4.github.io/CARET_doc/latest/configuration/inter_node_data_path/'
```
Comment (hsgwa) [inline src/caret_analyze/architecture/architecture.py:198]: nits
```suggestion
            'see https://tier4.github.io/CARET_doc/latest/configuration/inter_node_data_path/.'
```

**Review by hsgwa — COMMENTED** [review]
Just a minor comment. 

max_node_depth <= 15 case.
```
A search up to depth 10 has been completed. 
f the paths you want to measure cannot be found, consider specifying intermediate nodes. 
Also, if the number of paths is too large, consider filtering node/topic names. 
For details, see https://tier4.github.io/CARET_doc/latest/configuration/inter_node_data_path/.
```

max_node_depth > 15 case.
```
`max_node_depth` greater than 15 is not recommended because it significantly increases the search time and the number of returned paths.
**If the paths you want to measure cannot be found, consider specifying intermediate nodes.**
For details, see https://tier4.github.io/CARET_doc/latest/configuration/inter_node_data_path/.
// wait
A search up to depth 20 has been completed.
If the paths you want to measure cannot be found, consider specifying intermediate nodes.
Also, if the number of paths is too large, consider filtering node/topic names. For details, see https://tier4.github.io/CARET_doc/latest/configuration/inter_node_data_path/.
```
Since it is before the search, the sentence `If the paths you want to measure cannot be found` should be changed as follows.
```
`max_node_depth` greater than 15 is not recommended because it significantly increases the search time and the number of returned paths.
**If you are searching for paths that exceeds the max_node_depth, consider specifying an intermediate node. **
For details, see https://tier4.github.io/CARET_doc/latest/configuration/inter_node_data_path/.
// wait
A search up to depth 20 has been completed.
If the paths you want to measure cannot be found, consider specifying intermediate nodes.
Also, if the number of paths is too large, consider filtering node/topic names. For details, see https://tier4.github.io/CARET_doc/latest/configuration/inter_node_data_path/.
```

**Comment by atsushi421** [comment]
@hsgwa 
Thank you very much for pointing that out.
I have modified the output message, could you please review this again?

## Note
There was a discussion about renaming the argument `max_node_depth`, but I couldn't coming up with a good name, so I thought it would be better to add a description of the argument in docstring without renaming.
The docstring is not added in this PR because it was already covered in https://github.com/tier4/CARET_analyze/pull/195.

**Review by hsgwa — APPROVED** [review]
LGTM!

**Pull request merged**